packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -202,6 +202,13 @@ export interface LogConfig {
    * @default - Use AppSync default
    */
   readonly fieldLogLevel?: FieldLogLevel;
+
+  /**
+   * The role arn for CW Logs
+   *
+   * @default - None
+   */
+  readonly roleArn?: string;
```

Comment:
I think we should strongly type this for `iam.IRole`..

wdyt @MrArnoldPalmer

Reply:
Since we only need the `roleArn` as a string to be specified in `LogConfigProperty` as per [docs](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-appsync.CfnGraphQLApi.LogConfigProperty.html), I think it might be slightly weird to expect an impl of `IRole`.

The user might run into typed warnings for the object that (s)he is trying to pass too.

Reply:
CDK generally leans more towards strongly typing properties. 

[Design Guidelines](https://github.com/aws/aws-cdk/blob/master/DESIGN_GUIDELINES.md#types)

But @MrArnoldPalmer has final say

Reply:
Generally L2 constructs like to accept other constructs as args. For example, [`graphqlApi.addDynamoDbDataSource`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-appsync.GraphqlApi.html#add-wbr-dynamo-wbr-db-wbr-data-wbr-sourceid-table-options-span-class-api-icon-api-icon-experimental-title-this-api-element-is-experimental-it-may-change-without-notice-span) accepts an `ITable` as the argument and not just `tableName` or `tableArn`.

This is useful in the case that we may need additional properties or information about this role in the future and we don't want to incur breaking changes. Additionally we may want to add permissions to the role or otherwise mutate it.

So I'd agree that this should accept `IRole`.